transport/http/middleware_capture_request.go

```diff
@@ -0,0 +1,46 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/smithy-go/middleware"
+	"net/http"
+	"strconv"
+)
+
+const captureRequestID = "CaptureProtocolTestRequest"
+
+// AddCaptureRequestMiddleware captures serialized http request during protocol test for check
+func AddCaptureRequestMiddleware(stack *middleware.Stack, req *http.Request) error {
```

Comment:
**Question:** Do we _have_ to expose this in the public API?

If I understand correctly, the generated protocol tests in the SDK repo will have to see it to use it, but I think I'd rather we just generate it as unexported code in the protocol test instead of having it be exported in smithy-go just to be used for testing.

Reply:
Alternatively we can put it somewhere under `private/` like we've done in a few other places - to have it usable by the SDK, but to signal that we don't want other people doing so.

Reply:
I prefer alternative method cause the it prevent smithy go codegen using some sdk dependency of middleware. I will create a new `private/` directory under smithy-go repo and move the middleware there.